### PR TITLE
PXF & Erle-Brain 1.1 - battery voltage support

### PR DESCRIPTION
**Pull request by vmayoral**
This PR takes the work from https://github.com/diydrones/ardupilot/pull/2722 and rebases everything into a new branch that fixes a [reported](http://forum.erlerobotics.com/t/getting-feature-complete-ardupilot-to-work-on-erle-brain-v1-1/602/2) issue affecting the external compass.

TODO:
- [x] confirm that these changes do provide the expected functionality
- [ ] take into account the previous comments from @lucasdemarchi and address them.

### CODE REVIEW

**Comment by lucasdemarchi** [comment]
@vmayoral are you going to address those comments soon?

**Comment by vmayoral** [comment]
@lucasdemarchi in about a week or two. I'm fine with your comments though
so feel free to make the changes.


2016-01-01 22:16 GMT+01:00 Lucas De Marchi <notifications@github.com>:

**Comment by lucasdemarchi** [comment]
@vmayoral I'm probably going to wait your changes.

**Comment by vmayoral** [comment]
@lucasdemarchi can you have a look at those commits above? The aim is to address your first three comments.

Code still needs to be tested in hardware. Not to be merged yet.

File: libraries/AP_HAL_Linux/Storage.h [hunk 1]
```diff
@@ -2,7 +2,8 @@
 #define __AP_HAL_LINUX_STORAGE_H__
 
 #if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_ERLEBOARD || CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_PXF
-#define LINUX_STORAGE_USE_FRAM 1
+#define LINUX_STORAGE_USE_FRAM 0
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/Storage.h]: What's this? Yesterday I was using @tridge's pxf board and it was complaining about not being able to read FRAM... Is this something we should have a PR for?

**Comment by lucasdemarchi** [comment]
@vmayoral it's looking good. Thanks for addressing the comments. Later we will need to let this a bit more generic, removing the ifdefs, discovering in runtime the number of input sources, etc. But this can be done as improvement on top.

If this doesn't affect badly the other boards we can put it in and improve on top if you prefer.

File: libraries/AP_HAL_Linux/Storage.h (hunk 1 above)
Comment (vmayoral) [inline libraries/AP_HAL_Linux/Storage.h]: Thanks for pointing it out. It'll probably make sense to have it in a different PR. I'll look at it this weekend.

Comment (vmayoral) [inline libraries/AP_HAL_Linux/Storage.h]: A quick fix as a separate commit (on top of this one) has been added in another branch https://github.com/erlerobot/ardupilot/commit/b812b44e9ee543cae585085ea95dde0ee621d1b4.

I may have another look in the future but for now I'm most concerned about getting this existing PR into master.

**Comment by roger-zhao** [comment]
Hi, @vmayoral , "This PR takes ... fixes a reported issue affecting the external compass.", from my side, there's nothing related with this issue from commits yet, right?

**Comment by vmayoral** [comment]
@lucasdemarchi we've confirmed that these commits are valid for all the vehicles. It should be good to go.

Hi @roger-zhao, sorry, would you mind getting me in context?. I don't understand what you mean.

**Comment by lucasdemarchi** [comment]
@vmayoral see #3475

**Comment by vmayoral** [comment]
@lucasdemarchi good on me. If that gets in, the last commit in this PR will need a rebase. Let me know if you'd like me to go to ahead with that.

**Comment by lucasdemarchi** [comment]
@vmayoral I went ahead and did that. Just pushed this reviewed branch: https://github.com/lucasdemarchi/ardupilot/tree/review/pr-3406

I did some additional renames on top. Let me know what you think.

**Comment by lucasdemarchi** [comment]
Applied.

**Pull request closed without merging**